### PR TITLE
Add a default value for disable-mea-culpa-retries

### PR DESCRIPTION
**Pull request by pschorf**

### CODE REVIEW

**Comment by wyegelwel** [comment]
Lgtm though agree on integration comment

**Comment by pschorf** [comment]
Added a test: also, currently submit_time can be nil, but the schema didn't allow it (which would cause the same error in prod)

I made the schema more inclusive, let me know what you think.

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -265,7 +265,7 @@
       (merge {:framework-id (s/maybe s/Str)
               :status s/Str
               :state s/Str
-              :submit-time PosInt
+              :submit-time (s/maybe PosInt)
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: Why wouldn't submit-time be set?

File: scheduler/src/cook/mesos/api.clj [hunk 2]
```diff
@@ -741,7 +741,7 @@
                  :max_retries (:job/max-retries job) ; consistent with input
                  :retries_remaining (- (:job/max-retries job) (util/job-ent->attempts-consumed db job))
                  :max_runtime (:job/max-runtime job Long/MAX_VALUE) ; consistent with input
-                 :framework_id fid
+                 :framework_id (str fid)
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: why do we need a str here?

File: scheduler/src/cook/mesos/api.clj (hunk 1 above)
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: There's a comment:

; due to a bug, submit time may not exist for some jobs

In that case, as it stands you can't query those jobs. To me, I'd rather update the schema. We can also populate a default value instead.

File: scheduler/src/cook/mesos/api.clj [hunk 3]
```diff
@@ -729,7 +729,7 @@
         application (:job/application job)
         expected-runtime (:job/expected-runtime job)
         job-map {:command (:job/command job)
-                 :uuid (str (:job/uuid job))
+                 :uuid (:job/uuid job)
```
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: The schema specifies a UUID, not a string. I am guessing there is some magic in compojure that cleans this up but I'd prefer it to conform to the schema here.

File: scheduler/src/cook/mesos/api.clj (hunk 2 above)
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: The schema specifies a string, not a framework id record

Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: Actually, I don't think `str` is the right thing. Let me check this first before we merge.

Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: Ok I gave up on this one. I have no idea how this is getting converted, so I made a special case in the test.

**Pull request merged**